packages/tss-client/src/client.ts

```diff
@@ -43,9 +43,13 @@ if (globalThis.js_read_msg === undefined) {
             clearInterval(timer);
             resolve(found.msg_data);
           }
-          if (counter >= 500) {
+          if (counter >= 3000) {
             clearInterval(timer);
-            reject(new Error("Message not received in a reasonable time"));
+            // TODO Fix wasm to handle error objects properly and then reject
+            // with Error instead of string.
+            //
+            // eslint-disable-next-line prefer-promise-reject-errors
+            reject("Message not received in a reasonable time");
```

Comment:
Previous code was correct. Internally it is doing a map_err()

Reply:
I'm sorry to say that, but you are just wrong with that comment.

```
err.as_string().unwrap_or_else(|| "unknown cause".to_string()),
```
This is what Rust does. It tries to interpret the `err` as a string. This fails, if it is not a string, and hence it always currently maps to `"unknown cause"`.

The quick fix is to reject with a string error message. The proper fix is to change the Rust code, which, however, is more involved.

Instead of just saying that my code is incorrect, why don't you test your hypothesis first?

Reply:
The interface is defined as `Result<JsValue, JsValue>`. JsValue !== String. JsValue is actually a table.

The previous code here is correct. My previous comment told you where to implement the correct fix, basically JsValue <-> DKLSError prior to the map_err().

Reply:
This is the function we are using in Rust:

https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen/struct.JsValue.html#method.as_string

> If this JS value is a string value, this function copies the JS string value into wasm linear memory, encoded as UTF-8, and returns it as a Rust String.

I don’t know why you don’t believe that this works? I’ve tested this. You can test yourself.

Reply:
Can you state exactly what you think this should look like with a code example?

Reply:
As per the discussion, this can go in temporarily, please follow it up with the full fix.